src/hooks/useFirestoreDocWithAtom.ts

```diff
@@ -148,6 +149,13 @@ export function useFirestoreDocWithAtom<T = TableRow>(
           }
         }
 
+        // using `setDataAtom` here we are quickly changing state to instantly update the UI
+        // after that on actual update `setDataAtom` called once again with the final data
+        // the fix is mainly to fix the flicker effect on column reordering
+        setDataAtom((prev) => {
```

Comment:
https://github.com/rowyio/rowy/blob/2f8a3144223aee4ff986148596d9299c23ada566/src/hooks/useFirestoreDocWithAtom.ts#L78-L84

As shown in the first `useEffect`, we set `dataAtom` with a Promise to achieve React suspend. That was the root cause of why it got stuck with a loading spinner when we pushed this code to the staging branch. 

Solution is simple, just add a guard here to not set `dataAtom` in cases we want to suspense:
```typescript
if (disableSuspense) {
   setDataAtom((prev) => ({ ...prev, ...updateToDb }));
}
```

Reply:
@htuerker, thanks for the explanation.